src/engine/threaded_engine_perdevice.cc

```diff
@@ -244,6 +244,11 @@ class ThreadedEnginePerDevice : public ThreadedEngine {
     OprBlock* opr_block;
     RunContext run_ctx{ctx, stream};
     auto* task_queue = &(block->task_queue);
+
+    // Don't eat up omp threads for GPU jobs.  They're probably best used elsewhere,
+    // for example for image decoding or the optimizer pass
+    OpenMP::Get()->on_start_worker_thread(false);
```

Comment:
GPU worker doesn't use omp at all right? They only launch cuda kernels

Reply:
It shouldn't. Just in case it comes across some code that tries to use OMP, I think it would be better to not use OMP since the overhead probably wouldn't be worth it, which is why I explicitly set it to 1 in this case (no omp).

Reply:
Note: set__nproc(thread, new_nth)

```cpp
#define set__nproc(xthread, xval)                                              \
  (((xthread)->th.th_current_task->td_icvs.nproc) = (xval))

/* Changes set_nproc */
void __kmp_set_num_threads(int new_nth, int gtid) {
  kmp_info_t *thread;
  kmp_root_t *root;

  KF_TRACE(10, ("__kmp_set_num_threads: new __kmp_nth = %d\n", new_nth));
  KMP_DEBUG_ASSERT(__kmp_init_serial);

  if (new_nth < 1)
    new_nth = 1;
  else if (new_nth > __kmp_max_nth)
    new_nth = __kmp_max_nth;

  KMP_COUNT_VALUE(OMP_set_numthreads, new_nth);
  thread = __kmp_threads[gtid];

  __kmp_save_internal_controls(thread);

  set__nproc(thread, new_nth);

.
.
.
```